core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -192,21 +219,41 @@ private TrustAnchor getTrustAnchorForPublicKey(final JsonObject keyProps) {
             if (subjectDn == null || encodedKey == null) {
                 return null;
             } else {
+                final TrustAnchor trustAnchor;
                 try {
                     final String type = getProperty(keyProps, TenantConstants.FIELD_PAYLOAD_KEY_ALGORITHM, String.class,
                             "RSA");
                     final X509EncodedKeySpec keySpec = new X509EncodedKeySpec(Base64.getDecoder().decode(encodedKey));
                     final KeyFactory factory = KeyFactory.getInstance(type);
                     final PublicKey publicKey = factory.generatePublic(keySpec);
-                    return new TrustAnchor(subjectDn, publicKey, null);
+                    trustAnchor = new TrustAnchor(subjectDn, publicKey, null);
                 } catch (final IllegalArgumentException | GeneralSecurityException e) {
                     // Base64 decoding failed
                     return null;
                 }
+
+                final Boolean autoProvisioningEnabled = getProperty(keyProps,
+                        TenantConstants.FIELD_AUTO_PROVISIONING_ENABLED, Boolean.class, false);
+                if (autoProvisioningEnabled) {
+                    autoProvisioningEnabledIssuerDNs.add(subjectDn);
+                }
+                return trustAnchor;
```

Comment:
I do not really like the interlocking of the *trustAnchors* and the *autoProvisioningEnabledIssuerDNs* fields created by your changes. IMHO it makes the code hard(er) to understand and maintain.
Given that we are dealing with a few (< 10) trust anchors anyway, I think that it would be perfectly fine to iterate over the JSON objects in order to determine if a trust anchor may be used for auto-provisioning. WDYT?

Reply:
I understand that you want to separate the _trustAnchors_ from the flag. But it still makes sense to build up such a map (separately from the parsing of trust anchors), doesn't it? As I understand the TenantObject will be cached, so the same instance can be used for many authentication requests (potentially of different devices of the same tenant) and there is no need to parse the same json again and again. WDYT?

Reply:
I do not believe that there is much *parsing* involved as a `JsonObject` is basically a Map of Maps ... I doubt that there would be much of a performance difference ...